src/main/java/seedu/address/model/person/MatchingTagPredicate.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.model.person;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person}'s {@code Tag} matches the tag given.
+ */
+public class MatchingTagPredicate implements Predicate<Person> {
+
+    private final String tag;
+
+    public MatchingTagPredicate(String tag) {
+        this.tag = tag;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return ('[' + tag + ']').equals(person.getTag().toString());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof MatchingTagPredicate)) {
+            return false;
+        }
+
+        MatchingTagPredicate otherMatchingTagPredicate = (MatchingTagPredicate) other;
+        return tag.equals(otherMatchingTagPredicate.tag);
+    }
+
+    @Override
+    public String toString() {
+        return "tag " + tag;
+    }
+}
```

Comment:
I like how you separated the matching tags as a class to provide a clearer and efficient implementation.
Looks Good 👍🏾